### ISSUE TITLE
Bug: Z-Offset Bugs in the Artisan

### ISSUE THREAD

**Issue by mmack** [Snapmaker/Snapmaker-Artisan-Firmware-Issues#2]
I think there are 2 Bugs in the Artisan Firmware when it comes to auto-leveling:

**Problem 1**: It feels like the auto bed leveling does not take different zones into account. Or is not working at all. If i do a very height first layer (0.32mm) i can avoid scratching but still in some zones, especially outer ones, i see the line is really pressed into the bed. However if you look at the auto-leveling data you see the max height of the bed offset is 0.16mm. Why does a 0.32mm Line start scratching at all? I think auto-leveling is done but not "applied" to the g-code at all.

However, if i try to manually adjust the z-index while printing the first layer this happens:

**Problem 2**: If I do a z-index lizt of the nozzle while printing on the artisan's console the bed-leveling is not taken into account at all. I lift the nozzle for 0.5 mm immediately the nozzle beginns to scratch very bad. Before the lift it's only slightly getting worse, hence i wanna lift it manually.

I was not able to print on the very outside of the bed, so i scaled the model down. Originally it was 40x40cm.
On the images the outer parts work, in the inner zone it scratches...

I read out the auto-leveling matrix and also used Octoprint  for a Bed Visualizer Image:

``` 
Send: M420 V
Recv: Raw Bilinear Leveling Grid:
Recv:       0      1      2      3      4      5      6      7      8
Recv:  0 +17.107 +16.997 +16.878 +16.803 +16.737 +16.618 +16.533 +16.426 +16.323
Recv:  1 +17.185 +17.059 +16.938 +16.863 +16.778 +16.691 +16.567 +16.477 +16.386
Recv:  2 +17.248 +17.098 +17.030 +16.907 +16.843 +16.749 +16.627 +16.530 +16.441
Recv:  3 +17.248 +17.128 +17.033 +16.945 +16.875 +16.774 +16.670 +16.540 +16.471
Recv:  4 +17.267 +17.132 +17.041 +16.956 +16.886 +16.749 +16.674 +16.556 +16.480
Recv:  5 +17.215 +17.128 +17.045 +16.975 +16.892 +16.768 +16.630 +16.542 +16.431
Recv:  6 +17.212 +17.109 +17.029 +16.945 +16.846 +16.753 +16.611 +16.479 +16.394
Recv:  7 +17.117 +17.052 +16.973 +16.910 +16.816 +16.695 +16.565 +16.431 +16.329
Recv:  8 +17.018 +16.941 +16.913 +16.820 +16.721 +16.615 +16.451 +16.317 +16.184
Recv: 
Recv: compensated Bilinear Leveling Grid:
Recv:       0      1      2      3      4      5      6      7      8
Recv:  0 +17.614 +17.504 +17.385 +17.310 +17.244 +17.125 +17.040 +16.933 +16.830
Recv:  1 +17.692 +17.566 +17.445 +17.370 +17.285 +17.198 +17.074 +16.984 +16.893
Recv:  2 +17.755 +17.605 +17.537 +17.414 +17.350 +17.256 +17.134 +17.037 +16.948
Recv:  3 +17.755 +17.635 +17.540 +17.452 +17.382 +17.281 +17.177 +17.047 +16.978
Recv:  4 +17.774 +17.639 +17.548 +17.463 +17.393 +17.256 +17.181 +17.063 +16.987
Recv:  5 +17.722 +17.635 +17.552 +17.482 +17.399 +17.275 +17.137 +17.049 +16.938
Recv:  6 +17.719 +17.616 +17.536 +17.452 +17.353 +17.260 +17.118 +16.986 +16.901
Recv:  7 +17.624 +17.559 +17.480 +17.417 +17.323 +17.202 +17.072 +16.938 +16.836
Recv:  8 +17.525 +17.448 +17.420 +17.327 +17.228 +17.122 +16.958 +16.824 +16.691
Recv: Subdivided with CATMULL ROM Leveling Grid:
Recv:         0        1        2        3        4        5        6        7        8        9       10       11       12       13       14       15       16       17       18       19       20       21       22       23       24
Recv:  0 +17.61400 +17.57767 +17.54133 +17.50400 +17.46337 +17.42174 +17.38500 +17.35641 +17.33270 +17.31000 +17.28930 +17.26959 +17.24400 +17.20698 +17.16407 +17.12500 +17.09513 +17.06909 +17.04050 +17.00620 +16.96935 +16.93300 +16.89834 +16.86417 +16.83000
Recv:  1 +17.64074 +17.60257 +17.56439 +17.52552 +17.48371 +17.44120 +17.40404 +17.37602 +17.35334 +17.33059 +17.30717 +17.28369 +17.25657 +17.22315 +17.18609 +17.14989 +17.11639 +17.08373 +17.05089 +17.01735 +16.98362 +16.95013 +16.91709 +16.88428 +16.85148
Recv:  2 +17.66748 +17.62746 +17.58744 +17.54704 +17.50405 +17.46067 +17.42307 +17.39563 +17.37397 +17.35118 +17.32505 +17.29778 +17.26915 +17.23932 +17.20811 +17.17478 +17.13764 +17.09837 +17.06127 +17.02850 +16.99790 +16.96726 +16.93585 +16.90440 +16.87296
Recv:  3 +17.69250 +17.65011 +17.60772 +17.56600 +17.52372 +17.48211 +17.44550 +17.41737 +17.39424 +17.37000 +17.34228 +17.31344 +17.28450 +17.25711 +17.22961 +17.19800 +17.15833 +17.11455 +17.07450 +17.04206 +17.01333 +16.98450 +16.95424 +16.92387 +16.89350
Recv:  4 +17.71700 +17.67050 +17.62399 +17.58102 +17.54343 +17.50938 +17.47682 +17.44519 +17.41505 +17.38607 +17.35873 +17.33253 +17.30570 +17.27894 +17.25155 +17.21965 +17.17909 +17.13403 +17.09307 +17.06050 +17.03202 +17.00350 +16.97351 +16.94346 +16.91341
Recv:  5 +17.73977 +17.68863 +17.63748 +17.59348 +17.56248 +17.53862 +17.51152 +17.47513 +17.43550 +17.40037 +17.37453 +17.35318 +17.32969 +17.30240 +17.27296 +17.23963 +17.19928 +17.15503 +17.11448 +17.08133 +17.05188 +17.02261 +16.99241 +16.96239 +16.93237
Recv:  6 +17.75550 +17.70230 +17.64909 +17.60500 +17.57809 +17.56030 +17.53650 +17.49750 +17.45250 +17.41400 +17.38945 +17.37139 +17.35000 +17.32193 +17.29052 +17.25600 +17.21648 +17.17385 +17.13400 +17.09950 +17.06778 +17.03700 +17.00687 +16.97768 +16.94850
Recv:  7 +17.75929 +17.70922 +17.65915 +17.61682 +17.58898 +17.56887 +17.54398 +17.50663 +17.46449 +17.42811 +17.40408 +17.38581 +17.36407 +17.33547 +17.30339 +17.26880 +17.23077 +17.19021 +17.15082 +17.11326 +17.07685 +17.04326 +17.01396 +16.98746 +16.96096
Recv:  8 +17.75603 +17.71168 +17.66733 +17.62769 +17.59642 +17.56987 +17.54174 +17.50817 +17.47302 +17.44155 +17.41783 +17.39779 +17.37448 +17.34508 +17.31240 +17.27798 +17.24206 +17.20438 +17.16574 +17.12434 +17.08197 +17.04479 +17.01660 +16.99359 +16.97059
Recv:  9 +17.75500 +17.71428 +17.67355 +17.63550 +17.60161 +17.57039 +17.54000 +17.50942 +17.47968 +17.45200 +17.42863 +17.40731 +17.38250 +17.35131 +17.31663 +17.28150 +17.24771 +17.21346 +17.17650 +17.13291 +17.08659 +17.04700 +17.01952 +16.99876 +16.97800
Recv: 10 +17.76267 +17.72035 +17.67803 +17.63878 +17.60422 +17.57271 +17.54246 +17.51269 +17.48417 +17.45739 +17.43470 +17.41374 +17.38796 +17.35281 +17.31283 +17.27513 +17.24422 +17.21558 +17.18278 +17.14001 +17.09307 +17.05298 +17.02570 +17.00525 +16.98482
Recv: 11 +17.77255 +17.72656 +17.68056 +17.63900 +17.60455 +17.57454 +17.54542 +17.51566 +17.48678 +17.45978 +17.43781 +17.41771 +17.39104 +17.35093 +17.30425 +17.26309 +17.23508 +17.21260 +17.18489 +17.14458 +17.09904 +17.05963 +17.03214 +17.01077 +16.98941
Recv: 12 +17.77450 +17.72748 +17.68046 +17.63850 +17.60476 +17.57607 +17.54800 +17.51869 +17.48998 +17.46300 +17.44120 +17.42113 +17.39350 +17.35037 +17.29968 +17.25600 +17.22813 +17.20726 +17.18150 +17.14367 +17.10094 +17.06300 +17.03465 +17.01107 +16.98750
Recv: 13 +17.76069 +17.71864 +17.67659 +17.63837 +17.60626 +17.57797 +17.55054 +17.52297 +17.49626 +17.47037 +17.44798 +17.42640 +17.39765 +17.35493 +17.30504 +17.26030 +17.22714 +17.19914 +17.16937 +17.13422 +17.09730 +17.06254 +17.03195 +17.00351 +16.97507
Recv: 14 +17.73893 +17.70451 +17.67009 +17.63752 +17.60761 +17.57955 +17.55268 +17.52731 +17.50312 +17.47857 +17.45503 +17.43112 +17.40118 +17.36081 +17.31440 +17.26954 +17.22832 +17.18865 +17.15174 +17.11928 +17.08957 +17.05879 +17.02530 +16.99072 +16.95615
Recv: 15 +17.72250 +17.69315 +17.66379 +17.63500 +17.60670 +17.57896 +17.55250 +17.52841 +17.50559 +17.48150 +17.45639 +17.43000 +17.39900 +17.36126 +17.31891 +17.27500 +17.22817 +17.17978 +17.13700 +17.10496 +17.07854 +17.04950 +17.01424 +16.97637 +16.93850
Recv: 16 +17.72098 +17.69082 +17.66066 +17.63141 +17.60341 +17.57631 +17.55000 +17.52520 +17.50118 +17.47558 +17.44796 +17.41875 +17.38687 +17.35223 +17.31490 +17.27409 +17.22667 +17.17576 +17.12993 +17.09397 +17.06308 +17.03145 +16.99686 +16.96151 +16.92617
Recv: 17 +17.72480 +17.69126 +17.65772 +17.62614 +17.59784 +17.57149 +17.54517 +17.51876 +17.49237 +17.46437 +17.43382 +17.40166 +17.36902 +17.33776 +17.30603 +17.26941 +17.22383 +17.17335 +17.12574 +17.08360 +17.04432 +17.00783 +16.97506 +16.94508 +16.91511
Recv: 18 +17.71850 +17.68370 +17.64890 +17.61650 +17.58818 +17.56226 +17.53600 +17.50898 +17.48163 +17.45250 +17.42028 +17.38628 +17.35300 +17.32332 +17.29435 +17.26000 +17.21609 +17.16679 +17.11850 +17.07191 +17.02631 +16.98600 +16.95415 +16.92757 +16.90100
Recv: 19 +17.69370 +17.66277 +17.63184 +17.60204 +17.57397 +17.54702 +17.52056 +17.49531 +17.47053 +17.44333 +17.41199 +17.37822 +17.34439 +17.31265 +17.28083 +17.24485 +17.20189 +17.15476 +17.10772 +17.06042 +17.01321 +16.97146 +16.93879 +16.91157 +16.88435
Recv: 20 +17.65879 +17.63384 +17.60889 +17.58319 +17.55566 +17.52738 +17.50078 +17.47829 +17.45748 +17.43350 +17.40428 +17.37187 +17.33761 +17.30199 +17.26451 +17.22498 +17.18280 +17.13857 +17.09389 +17.04761 +17.00087 +16.95870 +16.92431 +16.89448 +16.86465
Recv: 21 +17.62350 +17.60232 +17.58113 +17.55850 +17.53291 +17.50587 +17.48050 +17.45954 +17.44024 +17.41750 +17.38947 +17.35798 +17.32350 +17.28550 +17.24450 +17.20250 +17.15998 +17.11646 +17.07250 +17.02696 +16.98098 +16.93850 +16.90200 +16.86900 +16.83600
Recv: 22 +17.59093 +17.56911 +17.54730 +17.52556 +17.50411 +17.48273 +17.46083 +17.43900 +17.41664 +17.39145 +17.36261 +17.33094 +17.29656 +17.25928 +17.21928 +17.17724 +17.13257 +17.08586 +17.03954 +16.99370 +16.94825 +16.90530 +16.86636 +16.82991 +16.79346
Recv: 23 +17.55796 +17.53331 +17.50865 +17.48678 +17.47086 +17.45772 +17.44066 +17.41671 +17.38882 +17.35922 +17.32865 +17.29635 +17.26228 +17.22721 +17.19037 +17.14937 +17.10143 +17.04934 +16.99902 +16.95261 +16.90798 +16.86465 +16.82289 +16.78244 +16.74198
Recv: 24 +17.52500 +17.49750 +17.47000 +17.44800 +17.43761 +17.43272 +17.42050 +17.39443 +17.36102 +17.32700 +17.29469 +17.26176 +17.22800 +17.19515 +17.16146 +17.12150 +17.07030 +17.01282 +16.95850 +16.91152 +16.86770 +16.82400 +16.77943 +16.73496 +16.69050
Recv: 
``` 

![newplot (1)](https://user-images.githubusercontent.com/100767/224480181-03052f46-cbc3-4227-8439-7a65d534c643.png)

![331746657_737405911181842_4264391441639108716_n](https://user-images.githubusercontent.com/100767/224480011-058dbe55-8842-4c21-97fd-abf99aaf832c.jpg)
![331782416_970512043834132_8521021309262021253_n](https://user-images.githubusercontent.com/100767/224480012-44f97c3b-a5a7-4df7-9bc8-7d3a8462ecc3.jpg)

**Comment by mmack** [comment]
And also: Every 10cm in printing (height) i need to manally raise the nozzle 0.05mm - 0.1mm to stop pushing the filament to heavy into the model... You can hear the difference pretty clear... the extruder starts beginning to click at some point if i do not raise the nozzle...

**Comment by mmack** [comment]
Some more testing for you: It does not matter if i do first the z-offset calibration and then the bed leveling, or the other way around. But watch the 2 attached videos from the same print. The z value does not change at all while printing the first layer...

https://user-images.githubusercontent.com/100767/226094740-3be60a1f-0809-4806-a9d7-24fd4b98cda1.mp4

https://user-images.githubusercontent.com/100767/226094744-15d8b0ab-3d6b-4539-9438-3d23caf578a7.mp4

**Comment by ganymedes** [comment]
I can confirm this issue, same happens with my artisan…!

**Comment by mmack** [comment]
https://github.com/MarlinFirmware/Marlin/issues/24337 Might be the actual problem. 
Would be nice to see someone from snapmaker looking at this problem.

**Comment by mmack** [comment]
I did a Bed Leveling with 80 degrees. Took one of the points, moved the printer to that position and queried the z-offset via M1006:

```
Recv: probed z: 17.055
Recv: Probing No. 52
Recv: x: 118.547249, y: 240.061859
[...]


Send: G0 X118.547249 Y240.061859 Z10
Recv: ok
[...]

Send: M1006
Recv: Tool Head: 3DP
Recv: fdm fault state: 0x0
Recv: fdm status: 2
Recv: live_z_offset  [0] = 0.000, [1] = 0.000
Recv: z compensation [0] = 0.507, [1] = 0.490

[...]
Recv: X:118.547 Y:240.062 Z:10.000 A:0.000 B:0.000 E:0.000 Count X:12012 Y:19412 Z:10998 A:0 B:0
Recv: ok



```

**Comment by mmack** [comment]
And here another point for the reference, should the compensation not be changed?
```
Recv: probed z: 16.870
Recv: Probing No. 74
Recv: x: 268.344238, y: 164.837128
[...]
Send: G0 X268.344238 Y164.837128 Z10
Recv: ok
[...]
Send: M1006
Recv: Tool Head: 3DP
Recv: fdm fault state: 0x0
Recv: fdm status: 2
Recv: live_z_offset  [0] = 0.000, [1] = 0.000
Recv: z compensation [0] = 0.507, [1] = 0.490
````

**Comment by ganymedes** [comment]
@sumibin any update on this issue?

**Comment by mmack** [comment]
So i do more live debugging here and note what i've found out. So the z compensation is the difference between the nozzle and the live_z_offset is the value from the artisan console if one adds manual z levels.
The question still remains: How to check if the bed leveling index is actually used at all...

**Comment by mmack** [comment]
And another finding:

I did the 3x3 Auto Leveling:
```
Recv: compensated Bilinear Leveling Grid:
Recv:       0      1      2
Recv:  0 +17.638 +17.234 +16.902
Recv:  1 +17.718 +17.397 +16.975
Recv:  2 +17.479 +17.204 +16.745
```

Then i manually raised the second point in the grid (it's the front midth of the bed) for 2mm with this command: 
`M421 I1 J0 Q2.00`

So the Bed nowlooks like this:
```
Recv: compensated Bilinear Leveling Grid:
Recv:       0      1      2
Recv:  0 +17.638 +19.234 +16.902
Recv:  1 +17.718 +17.397 +16.975
Recv:  2 +17.479 +17.204 +16.745
```

The calculated matrix looks like this:
```
Recv: Subdivided with CATMULL ROM Leveling Grid:
Recv:         0        1        2        3        4        5        6
Recv:  0 +17.63800 +18.31548 +18.99296 +19.23400 +18.74763 +17.82481 +16.90200
Recv:  1 +17.67648 +18.06412 +18.45175 +18.56078 +18.20545 +17.57150 +16.93756
Recv:  2 +17.71496 +17.81276 +17.91054 +17.88755 +17.66326 +17.31819 +16.97311
Recv:  3 +17.71800 +17.61474 +17.51148 +17.39700 +17.26382 +17.11941 +16.97500
Recv:  4 +17.66196 +17.50565 +17.34932 +17.21089 +17.10226 +17.01152 +16.92078
Recv:  5 +17.57048 +17.44990 +17.32931 +17.20744 +17.08345 +16.95817 +16.83289
Recv:  6 +17.47900 +17.39415 +17.30930 +17.20400 +17.06463 +16.90481 +16.74500
```` 

So what happens now if i print a line there: 
At the beginning of the line the nozzle nearly is scratching. In the middle it's lifting, but not 2mm and at the end of the line (most forward right point) it's nearly scratching again.
See this photo:
![334907186_905063677256142_3690333111319084361_n](https://user-images.githubusercontent.com/100767/227595447-ceb64b8e-2a28-46aa-a6a1-92caa2494669.jpg)

But also keep note the line next to it. It's the before-print extrusion line (or whatever it's called). There i could visibly see that it made a 2mm lift... the nozzle was away from the bed 2mm. On the photo that's the wobbled line. 


Any ideas on that?

**Comment by mmack** [comment]
I did it with the right nozzle. Only the right nozzle can use the complete Bed. Can you post your Bed Leveling Data here?

**Comment by ganymedes** [comment]
@mmack sorry it worked once, can't reproduce it... i give up, there is a bug in the firmware! Plz fix it snapmaker!

**Comment by KarstenSchwab** [comment]
Hi . Same Problem on my artisan. The kalibration is Not work . And also the kalibration is only in the middle and Not Nest enough To the outdide
![Uploading 8A5D76D5-71FB-482E-BE60-C26CF822CFF0.jpeg…]()

**Comment by KarstenSchwab** [comment]
![4B47261C-EFF8-4037-A163-110E6AE7DDF6](https://user-images.githubusercontent.com/99717259/227768798-08ad36ae-963a-4518-9eb3-a3f8ac7e17ea.jpeg)
Her
![D7CDEA9C-7CE9-4E75-86DC-6731C543AECE](https://user-images.githubusercontent.com/99717259/227768803-8b340970-fa13-4f70-857a-2df9030961ff.jpeg)
e some
![Uploading A7803623-11C1-42E6-881F-7145254C6DA6.jpeg…](
![3C1DB14F-C6A6-428C-8D1A-17122BE471F9](https://user-images.githubusercontent.com/99717259/227768811-285cc621-23d4-402c-ae3d-d7dba0a801c8.jpeg)
![CCEBBF41-774B-49D7-BDE8-3CFDA3B44008](https://user-images.githubusercontent.com/99717259/227768815-6f2cf17d-9a9a-4a12-9ec2-c45419c5e79d.jpeg)
![2069C331-2EB7-4B20-BDE4-32A8A6931315](https://user-images.githubusercontent.com/99717259/227768820-1e6f1d10-dcf2-4570-b364-acb854446643.jpeg)
![95489699-BD5F-480F-9FF5-197F8507BC9C](https://user-images.githubusercontent.com/99717259/227768835-2536f8ef-7066-4a19-9993-6581e631aa1e.jpeg)
![B40D06AF-45E7-415D-9B8D-1BFA4AEA19D5](https://user-images.githubusercontent.com/99717259/227768843-776d39d2-b414-4146-96ce-8e95823e8018.jpeg)
![D1381853-02F3-4C18-A52F-516FF3BE0BA2](https://user-images.githubusercontent.com/99717259/227768850-8ec94cba-500b-4d7b-89dd-4f09b22de5e4.jpeg)
![64A90639-547E-47C7-9133-837BB7874C9F](https://user-images.githubusercontent.com/99717259/227768857-57b09f59-f4e0-4857-b999-bcfe23901df0.jpeg)
![F0B8FEDD-D40C-4EFE-9549-48B21060D603](https://user-images.githubusercontent.com/99717259/227768873-b20c014e-e999-4ad8-87dd-47b32788b3e3.jpeg)
![E3F2F994-0593-4956-ADBA-3124125D5D30](https://user-images.githubusercontent.com/99717259/227768880-f47daac3-5b26-4552-985c-233e11ad65f4.jpeg)

**Comment by KarstenSchwab** [comment]
After all that kalibration i do my x x Test . On second try i do left 0,1 UP and
![Uploading E5E4CC28-468A-4F7D-8DC8-F1D674EBA16A.jpeg…]()
 Right 0.2 up 
![Uploading B4E0A14F-0D33-40BE-B94D-4FEEA5D80079.jpeg…]()

**Comment by KarstenSchwab** [comment]
After all the kalibration do x y . Not ok . I do left 0.1 up Right 0.2 
![Uploading 6A1BD264-019C-4530-9514-97D730D21587.jpeg…]()

**Comment by Th199403** [comment]
i have the same problem!  have tried everything

**Comment by jakubkinst** [comment]
Same problem here.

**Comment by Th199403** [comment]
I have the same issue and its been days and haven’t been able to resolve it. Can you please explain how you used a 0.2mm sheet to calibrate your Artisan?

I have logged a support ticket with Snapmaker; I am not at all pleased with their initial reply. Here is my reply, in the hopes someone in the forum can explain to me what I am doing wrong if any.

REPLY TO SNAPMAKER:

I am afraid the recommendation isn’t relevant as I have been using the in-built calibration method to try and get a good Z and X/Y axis result. When using the in-built calibration process the bed is heated to 70 degrees Celsius for the Bed-Levelling and 65 degrees Celsius for X /Y Calibration. I have run the calibration tests for hours at a time trying all manner of auto and manual calibration tests even manually offsetting to the axis to compensate but to no avail. So I am confident the bed has been appropriately heated.

For your reference I am running firmware 2.2.1.

The issue here isn’t the X/Y axis; yes it looks terrible but its because the Z axis is way off. What I have attempted to do over the last few days is:

METHOD 1

I have ran the 9 x 9 81 point bed-levelling calibration
Then ran the Z axis auto calibration.
Then the X/Y axis calibration ← When running this calibration test the dual print head moves to the bottom left of the build plate
to print the initial pre-test strips; the left print head (Snapmaker Black PLA) hardly prints any plastic at all; in fact you can hear the brass head dragging over the glass PEI plate!
The right print head (Snapmaker White Support PLA) begins to print its initial pre-test strip; it for the most part prints normally.
RESULT: The actual X/Y test is a failure; with the results appearing to show that the black PLA print head is running far too near the print bed. ← Well this is what I think, not speaking from experience though.
METHOD 2

I have ran the 9 x 9 81 point bed-levelling calibration
Then ran the Z axis Sensor calibration. ← Because the Z Axis Auto Calibration doesn’t seem to be working.
Using the supplied white calibration card; I step the heads down manually until both Right and Left heads are touching the card, can be pulled toward me but cannot be pushed back under the head it wrinkles.
I then run the Z axis Auto Calibration
Then the X/Y axis calibration ← When running this calibration test the dual print head moves to the bottom left of the build plate
to print the initial pre-test strips; the left print head (Snapmaker Black PLA) hardly prints any plastic at all; in fact you can hear the brass head dragging over the glass PEI plate!
The right print head (Snapmaker White Support PLA) begins to print its initial pre-test strip; it for the most part prints normally.
RESULT: Failure with the X/Y test looking no different to METHOD 1 and the print head dragging across the print bed
METHOD 3

I have ran the 9 x 9 81 point bed-levelling calibration
Then ran the Z axis Sensor calibration. ← Because the Z Axis Auto Calibration doesn’t seem to be working.
Using the supplied white calibration card; I step the heads down manually until both Right and Left heads are touching the card, can be pulled toward me but cannot be pushed back under the head it wrinkles.
I then run the Z axis Auto Calibration
Then the X/Y axis calibration ← When running this calibration test the dual print head moves to the bottom left of the build plate
to print the initial pre-test strips; the left print head (Snapmaker Black PLA) hardly prints any plastic at all; in fact you can hear the brass head dragging over the glass PEI plate!
The right print head (Snapmaker White Support PLA) begins to print its initial pre-test strip; it for the most part prints normally.
RESULT: Failure with the X/Y test looking no different to METHOD 1 and the print head dragging across the print bed
METHOD 4

I have ran the 9 x 9 81 point bed-levelling calibration
Then ran the Z axis Sensor calibration. ← Because the Z Axis Auto Calibration doesn’t seem to be working.
Using the supplied white calibration card; I step the heads down manually until both Right and Left heads are touching the card, can be pulled toward me but cannot be pushed back under the head it wrinkles.
I then run the Z axis Auto Calibration
Then the X/Y axis calibration ← When running this calibration test the dual print head moves to the bottom left of the build plate
I manually update the Right and Left heads with an offset of anywhere 0.05mm all the way to 0.2mm
RESULT: Failure with the X/Y test. No combination of positive offset seems to satisfy this test; the patterns either print correctly on the far left and right sides in the X axis and not in the middle. The Y axis will print correctly the top half of the pattern and then not at all the bottom half. There doesn’t seem to be any happy medium (an offset) that is somewhere in the middle that results in a good print.
METHOD 5

Reset factory defaults – This wipes the newer firmware, and the device defaults the original firmware.
Follow the Out of the Box experience calibration method. I.e load filament → Level Bed → Calibrate.
Same result as METHOD 1
If I had to hazard a guess the Z Axis sensor that measures how far it is from the build plate is not functioning.

BTW at least two other owners of the Artisan have experienced the same issue; you can find their posts here. I tried using a 0.2mm Feeler Gauge with the manual Senor Calibration; this didn’t work for me.

Please advise on how to troubleshoot and fix this issue; as it stands my very expensive investment is completely unusable and isn’t making anything wonderful!

**Comment by KarstenSchwab** [comment]
Hi . Same Problem by my artisan. Were is the 9x9 calibration find ? Thanks

**Comment by ganymedes** [comment]
@mmack Any update on this case? Any success with the new test firmware SM3_V2.2.19_20230404.bin?

**Comment by mmack** [comment]
@ganymedes finally someone who responds. Can you please identify who you are and for who you are speaking?

Printed half a bed and it did not scratch the surface anymore. However i changed something on the quick release myself to release pressure on the bed. I can't say what helped now. 
But what i can say is that i still have a gap between the heated bed and the print bed. Snapmaker should send us new beds which are even...

**Comment by ganymedes** [comment]
@mmack Sorry for the confusion. I'm a customer like you are with the same issue. My first layer is ok on one side of the bed but scratching on the other. Bed levelling is not working correctly and it's very frustrating!

**Comment by ganymedes** [comment]
@mmack ok, I might know what's going on. Auto levelling is pushing the nozzle to hard against the build plate. Therefore the build plate is bending a little bit and the reading is wrong. I did a whole 5x5 manual bed levelling and now it's working. Unfortunately the build plate is not heating when doing the manual procedure, so it's still not perfect but ok. Can someone test it as and confirm? Also I noticed that the build plate is only touching on four points. Is that on your Artisan also the case?

**Comment by Alex0176** [comment]
i have exacly the same issue on my snapmaker 2.0 - i ordered the dual extruder module, because the leveling should be better, but its the same issue, since i get it from my kickstarter campane. 

![image](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/18588859/49b73a22-1db8-46d8-8253-3eebdd8a5cac)
![image](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/18588859/4ab0ae61-3b6b-493b-aa70-22b71bfb3d42)
![image](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/18588859/c168aa8a-4f93-4fc9-945d-cbe3a15ca2f0)
![image](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/18588859/73d5487c-981f-4d64-acc8-e4d1d8813f93)

**Comment by r0dh3773** [comment]
Same issue here on a Snapmaker 2.0 A350, upgraded the linear modules and dual extrusion print head yesterday and cant get any usable bed leveling with the ABL / Auto Z-Offset.

The firmware was updated to the latest after the upgrade and the printhead is recognized correctly and everything appears to be ok.

But when it starts printing the calibration model it scratches into the build plate.

It doesnt look like the ABL mesh is being used at all…

**Comment by WhiskeyBanger** [comment]
Im also having major issues with the leveling process. I spent all day trying to fix it with no end in sight. i did factory resets, tried everything. I hope snap maker fixes this issue.

**Comment by BrennonTWilliams** [comment]
I'm having the same problem on my Snapmaker Artisan.

In this thread, users have provided a massive amount of data and documentation for Snapmaker devs to resolve this common and serious bug. Most Product Managers would kill for users that would give this much detail on bugs. If Snapmaker could muster half the enthusiasm and drive of its users, then bugs like this wouldn't be killing its brand.

This is why I no longer recommend Snapmaker to colleagues and friends. Users here serve up a wealth of thorough and thoughtfully collected measurements, log files, images, and samples. The Snapmaker team can't be bothered to even acknowledge they've looked.

Snapmaker, you don't deserve your users. Do better.

**Comment by PEKuus** [comment]
Same problem here, tried auto, manual and sensor many time now. Nothing gave the correct results.

**Comment by Zophiel** [comment]
This is exactly what is happening to my Artisan for a long time. I always thought it was due to my lack of knowledge.

- It would scrape at the edges of the build plate
- It would scrape across the print
- It would knock over the print at higher ranges.

This is causing VERY visible print lines. A new print module caused the same results. 
I think there is a discrepancy in the way it calculates the height and that accumulates the higher the print goes. Pressing into the 3D print.

**Comment by Klabusterbaer** [comment]
Hello,

I have exactly the same problems. The automatic bed calibration doesn't really work. I also don't understand why Snapmaker can't manage to change the calibration from 350x350 to the 400x400 from the Artisan. It's as if you can only use 80m2 of a 100m2 apartment because the landlord only has 80m2 in the contract.

![1713571144548](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/129724764/7200bc0b-04e5-4ce3-89cb-8b9e997eb3fd)

Here is an example after an automatic x/y and bed calibration.
![1713570838903](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/129724764/847a7e7a-be13-4669-ba71-b3b600c3744e)
![1713571369944](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/129724764/50a712e7-2d14-42b6-a49a-7d12b5358df2)

- I can only get halfway reasonable pressure when I adjust the bed manually (9x9).

- In terms of feeling, I also think that the settings change on their own for no apparent reason. With one print everything goes well and with the next one with the same settings, the nozzle scrapes over the heated bed.

- Just now when printing a skirt, the nozzle scratched over the bed for a very short time, then swung to the side and then moved back to its correct position and continued printing.
![1713570768001](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/129724764/aea3acbf-1162-4e07-8bf3-e816a7643dc7)

- It also happens again and again that the nozzle suddenly moves through the middle of the print despite the z-jump.
![1713571144536](https://github.com/Snapmaker/Snapmaker-Artisan-Firmware-Issues/assets/129724764/9ddf4e5f-349f-47c9-a78a-d2274d583a55)

- I also find the Snapmaker support video on YouTube about calibrating the sensor strange. It is not set there as described on the screen, but rather the calibration card must be able to be easily pulled out and pushed forward (without a fold).

Here is the YouTube link from the Snapmaker support video about sensor calibration:
[https://youtu.be/gqDGZTLWSPo?list=PLZiH-tfTScQJDgedKyLJDyj_SjvoTjl71](url)

ps: sorry for my bad google englisch :-)

**Comment by scotthsl** [comment]
Hello Mack and everyone who shared feedback on this issue, @mmack @ganymedes @Zophiel @Klabusterbaer @PEKuus @BrennonTWilliams @WhiskeyBanger @Th199403 @jakubkinst 
First, we want to thank you for your patience and detailed feedback. We recognize that our response here has been delayed, and we apologize for not addressing this issue sooner on GitHub. We did actively follow up on this issue through our support tickets, and since then, we have worked on specific firmware updates to address the auto-leveling challenges described here.

Issue Summary and Fixes
We found two key problems with the previous firmware. Here’s a breakdown of the solution:
1. Live Z Offset Issue & Fix
We identified a firmware bug that occasionally caused the Z-axis to compensate in the opposite direction when adjusting the live Z offset. For example, attempting to raise the nozzle sometimes resulted in downward pressure. This issue was fixed in Artisan Firmware v2.2.3 (released on March 31, 2023).
2. Segmented Compensation Optimization
During testing, we found a bug in segmented compensation functionality. With this bug resolved, segmented compensation now functions as expected. When the firmware receives a movement command in the XY plane that crosses multiple leveling grid points, the movement is now divided according to grid boundaries, applying multiple Z-axis compensation points throughout. This improvement provides better compensation during movement across the bed, helping to prevent inconsistent extrusion along a single line (over-tightening or loosening issues). This fix assumes that leveling data is accurate, without significant platform deformation. These updates were included in Artisan Firmware v2.2.4 (released on April 14, 2023).
Access all Artisan firmware downloads and updates here: https://forum.snapmaker.com/t/snapmaker-artisan-firmware-updates-and-downloads/29975

Further Troubleshooting Guidance
For anyone who continues to experience bed leveling issues even after updating to Firmware v2.2.4 (released on April 14, 2023) or later, we recommend checking the gap between the support platform and printing platform on your Artisan machine. In some cases, adding one or two shims (0.5–0.8 mm thick) between these two platforms would help resolve persistent leveling issues due to platform flatness. You can find additional troubleshooting steps here: [Snapmaker Troubleshooting Guide on First Layer Adhesion](https://wiki.snapmaker.com/en/general/troubleshooting/poor_first_layer_adhesion#the-support-platform-is-uneven-only-for-artisan).

We value the community's input, which has greatly assisted us in resolving these issues. If you have questions or continue to have bed leveling problems, contact the Snapmaker support team by submitting a ticket at https://snapmaker.formcrafts.com/support-ticket. The support team typically responds within two business days. Please include as much information and as many photos as possible in your submission to expedite the process.

Best regards,
The Snapmaker Teamg
